### ISSUE TITLE
Cannot find generated header files when building `unix-time` with Cabal 2.2

### ISSUE THREAD

**Issue by pechersky** [haskell/cabal#5169]
I have been trying to build hoogle using `cabal new-build hoogle`. It fails on building `unix-time` with `fatal error: config.h: No such file or directory`.

This is likely due to a missing `include-dirs: cbits` in the cabal file for `unix-time`. This was determined by a successful build of `kazu-yamamoto/unix-time` release `v0.3.7` by including that missing line. The build succeeded with `cabal new-build`.

This was found using cabal-install 2.2.0.0, ghc 8.0.2, WSL Ubuntu 16.04.

**Comment by hvr** [comment]
An important detail here is that this used to work in cabal-2.0

**Comment by hvr** [comment]
I was able to confirm my suspicion about #4874 being the culprit:

 - 783cbe6b86e7dbdcc5eaa8f7962caa574ef04fae exhibits the regression
 - whereas the parent commit c094940abc4266c8d013853b66271540cb2dc7f0 doesn't have the regression

Now the question is if we can recover support for old packages by extending the include header searchpath to the new locations, and/or whether this was an unintentional regression or an intentional one (in which case we may want to consider a legacy fallback for older `cabal-version:...` spec versions).

/cc @23Skidoo @angerman

**Comment by hvr** [comment]
While this doesn't make this issue go away, the author of `unix-time` mitigated this by via http://hdiff.luite.com/cgit/unix-time/diff?id=0.3.8&id2=0.3.7 (c.f. https://github.com/kazu-yamamoto/unix-time/pull/42)

**Comment by angerman** [comment]
I'm kind of on the fence to say this was broken to begin with...

**Comment by hvr** [comment]
I tend to agree w/ @angerman as I'm having a hard time coming up with a legacy fallback which doesn't disable new principled scheme from #4874 for legacy specVersions. Maybe we have to sacrifice the buildability of prior `unix-time` releases as there's not much we can do in a non-magical way IMO.

**Comment by RyanGlScott** [comment]
This also breaks the ability to `new-build` the `time` library with `cabal-install-2.2`, as documented in https://github.com/haskell/cabal/issues/5223#issue-306990907.

**Comment by gbaz** [comment]
`time` is so pervasive that breaking with that seems like it pushes up the need for a fix here...

**Comment by gbaz** [comment]
also as noted on that ticket "in the time case, the `include-dirs: lib/include` is there" so maybe its more possible to have a solution...

**Comment by hvr** [comment]
Fwiw, the issue in #5223 is in a different phase: This is about about `install-includes` referring to an auto-generated  header file which `lib:Cabal` is trying to locate (but looks in the wrong place), rather than an `-I` flag not being passed to GHC (and GHC complaining).

**Comment by hasufell** [comment]
Still a problem in 2020 building old unix-time.
